Documentation/Concepts/events.md

```diff
@@ -0,0 +1,133 @@
+
+![Flow of committing an event type](/images/concepts/eventtype.png)
+
+Event types are also important when wanting to deserialize events coming from other microservices. As the other microservice could be written in a completely different programming language, event types provide a level of abstraction for deserializing the events.
+
+{{< alert title="Why not use class/type names instead of GUIDs?" color="info" >}}
+When consuming events from other microservices it's important to remember that they name things according to their own domain and conventions.
+
+As an extreme example, a microservice could have an event with a type `CustomerRegistered`. But in another microservice in a different domain, written in a different language, this event type could be called `user_added`.
+
+GUIDs also solve the problem of having duplicate names, it's not hard to imagine having to have multiple events with the type of `CustomerRegisterd` in your code coming from different microservices.
+{{< /alert >}}
+
+#### Generations
+As the code changes, the event type is also bound to change at some point. These iterations on the same event type are called _generations_. Whenever you add or change a property in an event you should increase this number. This way the filters and handlers can handle a specific generation of an event.
+
+For example, imagine an older generation of an event type didn't have a property. You can have a separate handler for that particular generation that handles the missing property.
+```csharp
+// C# like pseudo-code
+[EventType("1844473f-d714-4327-8b7f-5b3c2bdfc26a", 1)]
+DishPrepared {
+    Dish string;
+}
+
+// Generation 2
+[EventType("1844473f-d714-4327-8b7f-5b3c2bdfc26a", 2)]
+DishPrepared {
+    Dish string;
+    Chef string;
+}
+
+// handlers generations separately
+[Handle("1844473f-d714-4327-8b7f-5b3c2bdfc26a", 1)]
+Handle(DishPrepared @event) {
+    // handle the missing property
+}
+
+[Handle("1844473f-d714-4327-8b7f-5b3c2bdfc26a", 2)]
+Handle(DishPrepared @event) { ... }
+```
+
+{{< alert title="Production ready" color="info" >}}
+For making development easier, you shouldn't worry about incrementing the generation until you're in production.
+{{< /alert >}}
+
+### Public vs. Private
+There is a basic distinction between private events and public events. In much the same way that you would not grant access to other applications to your internal database, you do not allow other applications to subscribe to your private events.
+
+Private events are only accessible within a single [Tenant]({{< ref "tenant" >}}) so that an event committed for one tenant cannot be handled outside of that tenant. Private events model the system within its domain and should be named in a way that makes sense inside the domain.
```

Comment:
I guess that the last sentence here is true for both private and public events? That's at least my approach to this. I would suggest just removing the last sentence.
```suggestion
Private events are only accessible within a single [Tenant]({{< ref "tenant" >}}) so that an event committed for one tenant cannot be handled outside of that tenant.
```